rosbag2/src/rosbag2/sequential_reader.cpp

```diff
@@ -71,9 +76,34 @@ SequentialReader::open(
   }
 }
 
+std::string SequentialReader::get_next_file()
+{
+  {
+    assert(current_file_iterator_ != file_paths_.end());
```

Comment:
You can store the end iterator in `open` instead of `file_paths_` since you only ever use `file_paths_` if you are comparing to the end iterator.